envelope.go

```diff
@@ -227,6 +232,11 @@ func (r *envelopeReader) Read(env *envelope) *Error {
 		for remaining > 0 {
 			bytesRead, err := io.CopyN(env.Data, r.reader, remaining)
 			if err != nil && !errors.Is(err, io.EOF) {
+				situation := fmt.Sprintf("read %d byte message", size)
```

Comment:
nit: you're possibly throwing out the construction of `situation`

Reply:
Meaning that we may be allocating a string but not using it?

Reply:
Yep! It's probably not important given this error path should be an uncommon event.

Reply:
Gotcha, good point - easy enough to fix!